### PR TITLE
Add Cancellation support to helix tasks and log console output uri for failed work items

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

**Comment by ChadNedzlek** [comment]
>                 for (; ; await Task.Delay(10000)) // delay every time this loop repeats

We should pass the token here too, and ideally the other awaited method too.

---
Refers to: src/Microsoft.DotNet.Helix/Sdk/CheckHelixJobStatus.cs:59 in 48c649d. [](commit_id = 48c649d941e5cc8e38db31c91769ac235556fbf1, deletion_comment = False)

**Comment by ChadNedzlek** [comment]
>             var results = await HelixApi.Aggregate.JobSummaryAsync(

I assume this thing can accept a cancellation token.

---
Refers to: src/Microsoft.DotNet.Helix/Sdk/CheckHelixJobStatus.cs:74 in 48c649d. [](commit_id = 48c649d941e5cc8e38db31c91769ac235556fbf1, deletion_comment = False)

**Comment by ChadNedzlek** [comment]
>                 ISentJob job = await def.SendAsync(msg => Log.LogMessage(msg));

I wonder if it's worth cancelling this.  It's going to have started the job, most likely, maybe it's worthwhile to wait on that request no matter what, so you have the correlation id's you need.

Ideally send it self would be cancellable, since it's zipping and uploading a lot of stuff, yes?

---
Refers to: src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs:237 in 48c649d. [](commit_id = 48c649d941e5cc8e38db31c91769ac235556fbf1, deletion_comment = False)

File: src/Microsoft.DotNet.Helix/Sdk/WaitForHelixJobCompletion.cs [hunk 1]
```diff
@@ -16,36 +18,39 @@ public class WaitForHelixJobCompletion : HelixTask
         [Required]
         public ITaskItem[] Jobs { get; set; }
 
-        protected override async Task ExecuteCore()
+        protected override async Task ExecuteCore(CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
             // Wait 1 second to allow helix to register the job creation
             await Task.Delay(1000);
```
Comment (ChadNedzlek) [inline src/Microsoft.DotNet.Helix/Sdk/WaitForHelixJobCompletion.cs]: More cancels!

**Comment by alexperovich** [comment]
It would be a bunch of work to add cancellation to the job sending part that I don't want to add to this PR. The part that takes the most time is waiting for the job anyways, and with this change it is cancelable.

**Comment by alexperovich** [comment]
@ChadNedzlek PTAL

**Pull request merged**